### PR TITLE
Harmonisation and expansion of github workflow

### PR DESCRIPTION
**Pull request by AliceBalfanz**
[Description of PR]
This PR only adjusts the github workflow. It harmonises the workflow based on https://github.com/dcs4cop/xcube-viewer/tree/master/.github/workflows and adds section to update k8s-configs for xcube-viewer app with the corresponding version in an automated manner. 
Checklist:

* ~~[ ] Add unit tests and/or doctests in docstrings~~
* ~~[ ] Add docstrings and API docs for any new/modified user-facing classes and functions~~
* ~~[ ] New/modified features documented in `docs/source/*`~~
* ~~[ ] Changes documented in `CHANGES.md`~~
* ~~[ ] AppVeyor CI passes~~
* ~~[ ] Test coverage remains or increases (target 100%)~~

Remember to close associated issues after merge!

### CODE REVIEW

File: .github/workflows/xcube_workflow.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ on:
   release:
     types: [published]
 
+env:
+  APP_NAME: xcube
+  ORG_NAME: bcdev
+  REG_NAME: quay.io
```
Comment (forman) [inline .github/workflows/xcube_workflow.yaml:11]:
```suggestion
  IMG_REG_NAME: quay.io
```

File: .github/workflows/xcube_workflow.yaml [hunk 2]
```diff
@@ -65,26 +67,23 @@ jobs:
         with:
           image: ${{ env.ORG_NAME }}/${{ env.APP_NAME }}
           tags: master, latest
-          registry: quay.io
-          username: ${{ secrets.QUAY_DOCKER_REPO_USERNAME }}
-          password: ${{ secrets.QUAY_DOCKER_REPO_PASSWORD }}
+          registry: ${{ env.REG_NAME }}
+          username: ${{ secrets.QUAY_REG_USERNAME }}
+          password: ${{ secrets.QUAY_REG_PASSWORD }}
```
Comment (forman) [inline .github/workflows/xcube_workflow.yaml:72]:
```suggestion
          registry: ${{ env.IMG_REG_NAME }}
          username: ${{ secrets.IMG_REG_USERNAME }}
          password: ${{ secrets.IMG_REG_PASSWORD }}
```

File: .github/workflows/xcube_workflow.yaml [hunk 3]
```diff
@@ -65,26 +67,23 @@ jobs:
...
         if: ${{ github.event_name == 'release' }}
         with:
           image: ${{ env.ORG_NAME }}/${{ env.APP_NAME }}
           tags: ${{ steps.release.outputs.tag }}
-          registry: quay.io
-          username: ${{ secrets.QUAY_DOCKER_REPO_USERNAME }}
-          password: ${{ secrets.QUAY_DOCKER_REPO_PASSWORD }}
+          registry: ${{ env.REG_NAME }}
+          username: ${{ secrets.QUAY_REG_USERNAME }}
+          password: ${{ secrets.QUAY_REG_PASSWORD }}
```
Comment (forman) [inline .github/workflows/xcube_workflow.yaml:82]:
```suggestion
          registry: ${{ env.IMG_REG_NAME }}
          username: ${{ secrets.IMG_REG_USERNAME }}
          password: ${{ secrets.IMG_REG_PASSWORD }}
```

File: .github/workflows/xcube_workflow.yaml [hunk 4]
```diff
@@ -105,7 +104,7 @@ jobs:
       - name: get-hash
         id: get-hash
         run: |
-          HASH=$(skopeo inspect docker://quay.io/bcdev/${{ env.APP_NAME }}:${{ steps.release.outputs.tag }} | jq '.Digest')
+          HASH=$(skopeo inspect docker://${{ env.REG_NAME }}/${{ env.ORG_NAME }}/${{ env.APP_NAME }}:${{ steps.release.outputs.tag }} | jq '.Digest')
```
Comment (forman) [inline .github/workflows/xcube_workflow.yaml:107]:
```suggestion
          HASH=$(skopeo inspect docker://${{ env.IMG_REG_NAME }}/${{ env.ORG_NAME }}/${{ env.APP_NAME }}:${{ steps.release.outputs.tag }} | jq '.Digest')
```

**Review by forman — CHANGES_REQUESTED** [review]
If you want to abstract from actual image registry in use, you should not use "QUAY" here: 
```
          username: ${{ secrets.QUAY_REG_USERNAME }}
          password: ${{ secrets.QUAY_REG_PASSWORD }}
```

In the long-term: Actually this is not a general xcube workflow for the official xcube docker image, because this is an application of xcube for BC. It should actually go into another, BC-internal repo.

**Review by TonioF — COMMENTED** [review]
Seconding @forman's suggestions

**Review by TonioF — APPROVED** [review]

**Review by forman — APPROVED** [review]

**Review by dzelge — APPROVED** [review]

**Pull request merged**